### PR TITLE
Feature/wish list filters

### PR DESCRIPTION
**Pull request by CowboyLynk**
#4 

- Added required courses functionality
- Removed DataTables javascript framework locally (it's on a CDN now)
- Added tooltips when hovering over a course
- Added a loading animation when loading all the content
- Stored wish list now remembers if the user marked the course as required

### CODE REVIEW

**Comment by DanielMcCrystal** [comment]
Lol you gotta put a description. For CodeU they made us put details of what we did, reasonings behind doing it the way we did it, and screenshots of the new features.

**Comment by CowboyLynk** [comment]
I did put a description. It says #4

**Comment by DanielMcCrystal** [comment]
The CSS looks awesome. But add a description

File: courses_database/static/js/wishlist.js [hunk 1]
```diff
@@ -104,9 +105,10 @@ class WishListItem{
         }
         button.appendChild(button_required);
         button.appendChild(button_text);
+        button.title = this.title;
         button.style.backgroundColor = color;
         button.style.color = text_color;
-        button.classList.add("wish-list-item");
+        button.classList.add("wish_list_item");
```
Comment (DanielMcCrystal) [inline courses_database/static/js/wishlist.js]: Where does it tell the button to display the title when hovered?

**Comment by CowboyLynk** [comment]
Added description

**Pull request merged**